### PR TITLE
Tweak the XAML generator

### PR DESCRIPTION
**Pull request by duncanp-sonar**
- alter whitespace handling
- add table cell style

### CODE REVIEW

File: src/Education/XamlGenerator/IRuleHelpXamlBuilder.cs [hunk 1]
```diff
@@ -217,14 +216,14 @@ private void ProcessElement()
 
                 case "p":
                     writer.WriteStartElement("Paragraph");
-                    writer.WriteAttributeString("xml", "space", null, "default");
```
Comment (duncanp-sonar) [inline src/Education/XamlGenerator/IRuleHelpXamlBuilder.cs:220]: I've made a few changes to how whitespace is being handled, having looked at some sample generated rule help.

Previously, whitespace was being `preserved` at the top-level, and turned off (`default`) on specific child cells i.e. most whitespace (particularly line breaks) would be rendered in the output. However, that was leading to line breaks appearing in odd places, so I've flipped it round.

Now, whitespace between tags is ignored (the default). It is only preserved for specific child elements, namely those inside the `pre` <html> tag.

File: src/Education/XamlGenerator/IRuleHelpXamlBuilder.cs [hunk 2]
```diff
@@ -297,10 +296,8 @@ private void ProcessElement()
                     writer.WriteStartElement("TableCell");
                     PushOutputElementInfo("td", false);
 
-                    if (tableAlternateRow)
-                    {
-                        ApplyStyleToStartElement(StyleResourceNames.TableBodyCellAlternateRow);
-                    }
+                    var cellStyle = tableAlternateRow ? StyleResourceNames.TableBodyCellAlternateRow : StyleResourceNames.TableBodyCell;
```
Comment (duncanp-sonar) [inline src/Education/XamlGenerator/IRuleHelpXamlBuilder.cs:299]: I also added a style for "normal" table cells. That means we can define one style for the odd-numbered rows, and another for the even-numbered rows.

Comment (bigfluffycookie) [reply src/Education/XamlGenerator/IRuleHelpXamlBuilder.cs:299]: Ah very nice!

File: src/Education/XamlGenerator/IRuleHelpXamlBuilder.cs (hunk 1 above)
Comment (bigfluffycookie) [reply src/Education/XamlGenerator/IRuleHelpXamlBuilder.cs:220]: Makes sense

**Review by bigfluffycookie — APPROVED** [review]
Looks good! :)

**Pull request merged**